[2025-07-12T15:50:19.660+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: flight_data_quality_check.check_data_validity manual__2025-07-12T15:50:19.294608+00:00 [queued]>
[2025-07-12T15:50:19.664+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: flight_data_quality_check.check_data_validity manual__2025-07-12T15:50:19.294608+00:00 [queued]>
[2025-07-12T15:50:19.664+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 2
[2025-07-12T15:50:19.670+0000] {taskinstance.py:2191} INFO - Executing <Task(PythonOperator): check_data_validity> on 2025-07-12 15:50:19.294608+00:00
[2025-07-12T15:50:19.673+0000] {standard_task_runner.py:60} INFO - Started process 1094 to run task
[2025-07-12T15:50:19.675+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'flight_data_quality_check', 'check_data_validity', 'manual__2025-07-12T15:50:19.294608+00:00', '--job-id', '124', '--raw', '--subdir', 'DAGS_FOLDER/flight_data_quality.py', '--cfg-path', '/tmp/tmps1h6mk5y']
[2025-07-12T15:50:19.677+0000] {standard_task_runner.py:88} INFO - Job 124: Subtask check_data_validity
[2025-07-12T15:50:19.697+0000] {task_command.py:423} INFO - Running <TaskInstance: flight_data_quality_check.check_data_validity manual__2025-07-12T15:50:19.294608+00:00 [running]> on host f1b3ce86e5d1
[2025-07-12T15:50:19.826+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='data_engineer' AIRFLOW_CTX_DAG_ID='flight_data_quality_check' AIRFLOW_CTX_TASK_ID='check_data_validity' AIRFLOW_CTX_EXECUTION_DATE='2025-07-12T15:50:19.294608+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-12T15:50:19.294608+00:00'
[2025-07-12T15:50:19.831+0000] {base.py:83} INFO - Using connection ID 'postgres_default' for task execution.
[2025-07-12T15:50:19.833+0000] {sql.py:450} INFO - Running statement: 
        SELECT flight_number, COUNT(*) as count
        FROM scheduled_flights
        GROUP BY flight_number
        HAVING COUNT(*) > 1
        LIMIT 10
        , parameters: None
[2025-07-12T15:50:19.834+0000] {sql.py:459} INFO - Rows affected: 0
[2025-07-12T15:50:19.835+0000] {logging_mixin.py:188} INFO - No duplicate flight numbers found
[2025-07-12T15:50:19.837+0000] {base.py:83} INFO - Using connection ID 'postgres_default' for task execution.
[2025-07-12T15:50:19.839+0000] {sql.py:450} INFO - Running statement: 
        SELECT code, latitude, longitude
        FROM airports
        WHERE latitude NOT BETWEEN -90 AND 90 
           OR longitude NOT BETWEEN -180 AND 180
        , parameters: None
[2025-07-12T15:50:19.839+0000] {sql.py:459} INFO - Rows affected: 0
[2025-07-12T15:50:19.840+0000] {logging_mixin.py:188} INFO - All airport coordinates are valid
[2025-07-12T15:50:19.843+0000] {base.py:83} INFO - Using connection ID 'postgres_default' for task execution.
[2025-07-12T15:50:19.844+0000] {sql.py:450} INFO - Running statement: 
        SELECT flight_number, origin_airport, destination_airport
        FROM scheduled_flights
        WHERE origin_airport = destination_airport
        LIMIT 5
        , parameters: None
[2025-07-12T15:50:19.845+0000] {sql.py:459} INFO - Rows affected: 0
[2025-07-12T15:50:19.845+0000] {logging_mixin.py:188} INFO - No flights with same origin and destination found
[2025-07-12T15:50:19.845+0000] {python.py:201} INFO - Done. Returned value was: None
[2025-07-12T15:50:19.849+0000] {taskinstance.py:1138} INFO - Marking task as SUCCESS. dag_id=flight_data_quality_check, task_id=check_data_validity, execution_date=20250712T155019, start_date=20250712T155019, end_date=20250712T155019
[2025-07-12T15:50:19.859+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 0
[2025-07-12T15:50:19.878+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
